src/qiskit_qec/utils/stim_tools.py

```diff
@@ -0,0 +1,347 @@
+            else:
+                # Gates and measurements
+                if inst.name in qiskit_to_stim_dict:
+                    if len(cargs) > 0:  # keeping track of measurement indices in stim
+                        measurement_data.append(
+                            [
+                                cargs[0]._index + register_offset[qargs[0]._register.name],
+                                qargs[0]._register.name,
+                            ]
+                        )
+                    if qiskit_to_stim_dict[inst.name] == "TICK":  # barrier
+                        stim_circuit.append("TICK")
+                    else:  # gates/measurements acting on qubits
+                        stim_circuit.append(qiskit_to_stim_dict[inst.name], qubit_indices)
+                else:
+                    raise Exception("Unexpected operations: " + str([inst, qargs, cargs]))
+
+        stim_circuits[circ_label] = stim_circuit
+        stim_measurement_data[circ_label] = measurement_data
+    return stim_circuits, stim_measurement_data
+
+
+def get_counts_via_stim(circuits, shots: int = 4000, noise_model=None):
+    """Returns a qiskit compatible dictionary of measurement outcomes
+
+    Args:
+        circuit: Qiskit circuit compatible with `get_stim_circuits` or list thereof.
+        shots: Number of samples to be generated.
+        noise_model: Pauli noise model for any additional noise to be applied.
+
+    Returns:
+        counts: Counts dictionary in standard Qiskit form or list thereof.
+    """
+
+    if noise_model:
+        circuits = noisify_circuit(circuits, noise_model)
+
+    single_circuit = isinstance(circuits, QuantumCircuit)
+    if single_circuit:
+        circuits = [circuits]
+
+    counts = []
+    for circuit in circuits:
+        stim_circuits, stim_measurement_data = get_stim_circuits({"": circuit})
+        stim_circuit = stim_circuits[""]
+        measurement_data = stim_measurement_data[""]
+
+        stim_samples = stim_circuit.compile_sampler().sample(shots=shots)
+        qiskit_counts = {}
+        for stim_sample in stim_samples:
+            prev_reg = measurement_data[-1][1]
+            qiskit_count = ""
+            for idx, meas in enumerate(measurement_data[::-1]):
+                _, reg = meas
+                if reg != prev_reg:
+                    qiskit_count += " "
+                qiskit_count += str(int(stim_sample[-idx - 1]))
+                prev_reg = reg
+            if qiskit_count in qiskit_counts:
+                qiskit_counts[qiskit_count] += 1
+            else:
+                qiskit_counts[qiskit_count] = 1
+        counts.append(qiskit_counts)
+
+    if single_circuit:
+        counts = counts[0]
+
+    return counts
+
+
+def detector_error_model_to_rx_graph(model: StimDetectorErrorModel) -> rx.PyGraph:
+    """Convert a stim error model into a RustworkX graph.
+    It assumes that the stim circuit does not contain repeat blocks.
```

Comment:
Do repeat blocks raise a NotImplementedError? If so, that's fine. Could you then add an issue for handling repeat blocks for the error_model_to_rx_graph?

Reply:
Yes. It is coming soon..